utils/rand/rand.go

```diff
@@ -6,13 +6,18 @@ import (
 	"fmt"
 )
 
-// This package is a wrppaer around true RNG crypto/rand.
-// It implements useful tools using the true RNG and that
-// are not exported by the crypto/rand package.
-// This package does not implement any determinstic RNG (Pseudo RNG)
-// unlike the package flow-go/crypto/random.
+// This package is a wrapper around `crypto/rand` that uses the system RNG underneath
+// to extract secure entropy.
+// It implements useful tools that are not exported by the `crypto/rand` package.
+// This package should be used instead of `math/rand` for any use-case requiring
+// a secure randomness. It provides similar APIs to the ones provided by `math/rand`.
+// This package does not implement any determinstic RNG (Pseudo-RNG) based on
+// user input seeds. For the deterministic use-cases please use `flow-go/crypto/random`.
 
-// returns a random uint64
+// Uint64 returns a random uint64.
+// It returns:
+//   - (0, err) if crypto/rand fails to provide entropy which is likely a result of a system error.
```

Comment:
```suggestion
//   - (0, exception) if crypto/rand fails to provide entropy which is likely a result of a system error.
```
I'd prefer to use the exception terminology to differentiate from eg. errors from validation functions. By returning an exception, we are requiring the caller to crash. 

Applies to all the functions in this file.

Reply:
Good suggestion.